examples/dao/test/failing-path.js

```diff
@@ -0,0 +1,585 @@
+    });
+  });
+
+  describe('Deposit Vote Token', function () {
+    this.beforeAll(() => {
+      ctx.addProposal();
+    });
+
+    const _depositVoteToken = (account, lsig, amount) => {
+      const depositVoteTx = mkDepositVoteTokenTx(
+        ctx.daoAppID, ctx.govTokenID, account, lsig, amount
+      );
+      ctx.executeTx(depositVoteTx);
+    };
+
+    it('should fail if voterAccount is not optedIn to DAO App', () => {
+      assert.throws(
+        () => _depositVoteToken(ctx.voterA.account, ctx.depositLsig, 6),
+        APP_NOT_FOUND
+      );
+    });
+
+    it('should fail if gov tokens deposit transaction is not present', () => {
+      ctx.optInToDAOApp(ctx.voterA.address);
+      ctx.optInToDAOApp(ctx.voterB.address);
+      ctx.syncAccounts();
+
+      const txParams = mkDepositVoteTokenTx(
+        ctx.daoAppID,
+        ctx.govTokenID,
+        ctx.voterA.account,
+        ctx.depositLsig,
+        6
+      );
+      assert.throws(
+        () => ctx.executeTx({ ...txParams[0] }),
+        INDEX_OUT_OF_BOUND_ERR
+      );
+    });
+
+    it('should fail if receiver is not depositLsig', () => {
+      assert.throws(
+        () => _depositVoteToken(ctx.voterA.account, ctx.daoFundLsig, 6),
+        RUNTIME_ERR1009
+      );
+    });
+  });
+
+  describe('Vote', function () {
+    this.beforeAll(() => {
+      ctx.depositVoteToken(ctx.voterA, ctx.depositLsig, 6);
+    });
+
+    let registerVoteParam;
+    this.beforeEach(() => {
+      registerVoteParam = {
+        type: types.TransactionType.CallApp,
+        sign: types.SignType.SecretKey,
+        fromAccount: ctx.voterA.account,
+        appID: ctx.daoAppID,
+        payFlags: { totalFee: 2000 },
+        appArgs: ['str:register_vote', `str:${Vote.YES}`],
+        accounts: [ctx.proposalLsig.address()]
+      };
+
+      // set "now" between [votingStart, votingEnd]
+      ctx.runtime.setRoundAndTimestamp(10, votingStart + ((votingEnd - votingStart) / 2));
+    });
+
+    it('should fail if proposalLsig address is not passed', () => {
+      assert.throws(
+        () =>
+          ctx.executeTx({
+            ...registerVoteParam,
+            accounts: [ctx.depositLsig.address()] // different address passed
+          }),
+        RUNTIME_ERR1009
+      );
+    });
+
+    it('should fail if no votes are deposited', () => {
+      // voterB hasn't deposited votes yet
+      assert.throws(
+        () => ctx.executeTx({ ...registerVoteParam, fromAccount: ctx.voterB.account }),
+        RUNTIME_ERR1009
+      );
+    });
+
+    it('should fail user tries to register votes again for same proposal (double voting)', () => {
+      // register votes by voterA
+      ctx.executeTx({ ...registerVoteParam, fromAccount: ctx.voterA.account });
+
+      // fails (voter A tries to register votes again)
+      assert.throws(
+        () => ctx.executeTx({ ...registerVoteParam, fromAccount: ctx.voterA.account }),
+        RUNTIME_ERR1009
+      );
+    });
```

Comment:
add the following scenario:
* user deposits gov tokens for voting (OK)
* user votes (OK)
* user deposits again (OK)
* user tries to vote again after depositing more tokens (fail)

Reply:
done